tests/instruments/hp/test_hp856Xx.py

```diff
@@ -0,0 +1,477 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+import pytest
+
+from pymeasure.test import expected_protocol
+
+from pymeasure.instruments.hp import HP8560A, HP8561B
+from pymeasure.instruments.hp.hp856Xx import Trace, MixerMode, CouplingMode, DemodulationMode, \
+    DetectionModes, AmplitudeUnits, HP856Xx, ErrorCode, FrequencyReference
+
+
+class TestHP856Xx:
+    def test_id(self):
+        with expected_protocol(
+                HP856Xx,
+                [("ID?", "HP8560A,002,H03")],
+        ) as instr:
+            assert instr.id == "HP8560A,002,H03"
+
+    def test_attenuation(self):
+        with expected_protocol(
+                HP856Xx,
+                [("AT 70", None),
+                 ("AT?", "70"),
+                 ("AT AUTO", None),
+                 ("AT?", "20"),
+                 ("AT 20", None),
+                 ("AT?", "20")],
+        ) as instr:
+            # test set and get of attenuation as integer
+            instr.attenuation = 70
+            assert instr.attenuation == 70
+            # test string parameters
+            instr.attenuation = "AUTO"
+            assert instr.attenuation == 20
+            # test truncation
+            instr.attenuation = 16
+            assert instr.attenuation == 20
```

Comment:
It is better to use pytests `parametrize` function if you test different values, that way you notice, if a single value fails or all values.